Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/App.tsx

```diff
@@ -1,25 +1,118 @@
+import { useState } from "react";
+
 import BatteryInput from "~/components/batteryInput";
 import Header from "~/components/header";
 import SpeedInput from "~/components/speedInput";
 import WeatherInput from "~/components/weatherInput";
 
 const App = () => {
-  const calculateRange = () => {
-    return;
-  }
+  const [speed, setSpeed] = useState<number | undefined>(0);
+  const [speedValid, setSpeedValid] = useState(true);
```

Comment:
we're using state to represent the valid state for battery and speed. do we need to create state for this?

remember that your app will re-render when any of the fields change, which is fine. But can we use that to our advantage and not create new state and make those "valid" variables a computed value from the state?